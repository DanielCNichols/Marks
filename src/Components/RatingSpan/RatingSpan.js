import React from 'react';
import { IoIosStar, IoIosStarOutline } from 'react-icons/io';
import './RatingSpan.css';

export default function RatingSpan(props) {
  let value = props.rating;
  let noStars = value;
  let noEmpty = 5 - value;
  let stars = [];

  if (!value) {
    return <span className="NA">Not Rated</span>;
  }

  for (let i = 0; i < noStars; i++) {
    stars.push(<IoIosStar className="filled" />);
  }

  for (let i = 0; i < noEmpty; i++) {
    stars.push(<IoIosStarOutline />);
  }

  return (
    <div className="rating">
      {stars.map((star, idx) => {
        return <span key={idx}>{star}</span>;
      })}
    </div>
  );
}
