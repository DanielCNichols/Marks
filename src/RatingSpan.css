.filled {
  color: gold;
}

.rating svg {
  font-size: 1.3em;
}
