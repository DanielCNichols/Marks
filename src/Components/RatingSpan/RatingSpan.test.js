import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RatingSpan from './RatingSpan';
import { BrowserRouter } from 'react-router-dom';

describe('LandingPage tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RatingSpan />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected when there is no value prop', () => {
    const tree = renderer.create(<RatingSpan />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected when there is a value prop', () => {
    const tree = renderer.create(<RatingSpan value={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
