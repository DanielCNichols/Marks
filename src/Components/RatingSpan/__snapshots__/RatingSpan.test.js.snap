// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage tests renders the UI as expected when there is a value prop 1`] = `
<span
  className="NA"
>
  Not Rated
</span>
`;

exports[`LandingPage tests renders the UI as expected when there is no value prop 1`] = `
<span
  className="NA"
>
  Not Rated
</span>
`;
